File: php/test/Creational/FactoryMethodTest.php
<?

require_once 'src/Creational/FactoryMethod/SoupFactoryMethod.php';
require_once 'src/Creational/FactoryMethod/BostonSoupFactoryMethodSubclass.php';
require_once 'src/Creational/FactoryMethod/HonoluluSoupFactoryMethodSubclass.php';

class Creational_FactoryMethodTest extends PHPUnit_Framework_TestCase
{
 
	function test()
	{

		$soupFactoryMethod = new SoupFactoryMethod();
		$soupBuffet = $soupFactoryMethod->makeSoupBuffet();
		$soupBuffet->soupBuffetName= $soupFactoryMethod->makeBuffetName();
		$soupBuffet->chickenSoup= $soupFactoryMethod->makeChickenSoup();
		$soupBuffet->clamChowder= $soupFactoryMethod->makeClamChowder(); 
		$soupBuffet->fishChowder= $soupFactoryMethod->makeFishChowder();
		$soupBuffet->minnestrone= $soupFactoryMethod->makeMinnestrone();
		$soupBuffet->pastaFazul= $soupFactoryMethod->makePastaFazul(); 
		$soupBuffet->tofuSoup= $soupFactoryMethod->makeTofuSoup();
		$soupBuffet->vegetableSoup= $soupFactoryMethod->makeVegetableSoup(); 
		$this->assertEquals("At the  " . $soupBuffet->soupBuffetName . $soupBuffet->getTodaysSoups(), 
			"At the  Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: ClanChowder Fish Chowder: FishChowder Minnestrone: Minnestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			'Testing plain soup buffet');


		$bostonSoupFactoryMethod = new BostonSoupFactoryMethodSubclass();
		$bostonSoupBuffet = $bostonSoupFactoryMethod->makeSoupBuffet();
		$bostonSoupBuffet->soupBuffetName= $bostonSoupFactoryMethod->makeBuffetName();
		$bostonSoupBuffet->chickenSoup= $bostonSoupFactoryMethod->makeChickenSoup();
		$bostonSoupBuffet->clamChowder= $bostonSoupFactoryMethod->makeClamChowder();
		$bostonSoupBuffet->fishChowder= $bostonSoupFactoryMethod->makeFishChowder();
		$bostonSoupBuffet->minnestrone= $bostonSoupFactoryMethod->makeMinnestrone();
		$bostonSoupBuffet->pastaFazul= $bostonSoupFactoryMethod->makePastaFazul();
		$bostonSoupBuffet->tofuSoup= $bostonSoupFactoryMethod->makeTofuSoup();
		$bostonSoupBuffet->vegetableSoup= $bostonSoupFactoryMethod->makeVegetableSoup(); 
		$this->assertEquals("At the  " . $bostonSoupBuffet->soupBuffetName . $bostonSoupBuffet->getTodaysSoups(), 
			"At the  Boston Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: QuahogChowder Fish Chowder: ScrodFishChowder Minnestrone: Minnestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			'Testing Boston soup buffet');

		$honoluluSoupFactoryMethod = new HonoluluSoupFactoryMethodSubclass();
		$honoluluSoupBuffet = $honoluluSoupFactoryMethod->makeSoupBuffet();
		$honoluluSoupBuffet->soupBuffetName= $honoluluSoupFactoryMethod->makeBuffetName();
		$honoluluSoupBuffet->chickenSoup= $honoluluSoupFactoryMethod->makeChickenSoup();
		$honoluluSoupBuffet->clamChowder= $honoluluSoupFactoryMethod->makeClamChowder(); 
		$honoluluSoupBuffet->fishChowder= $honoluluSoupFactoryMethod->makeFishChowder();
		$honoluluSoupBuffet->minnestrone= $honoluluSoupFactoryMethod->makeMinnestrone();
		$honoluluSoupBuffet->pastaFazul= $honoluluSoupFactoryMethod->makePastaFazul(); 
		$honoluluSoupBuffet->tofuSoup= $honoluluSoupFactoryMethod->makeTofuSoup();
		$honoluluSoupBuffet->vegetableSoup= $honoluluSoupFactoryMethod->makeVegetableSoup(); 
		$this->assertEquals("At the  " . $honoluluSoupBuffet->soupBuffetName . $honoluluSoupBuffet->getTodaysSoups(), 
			"At the  Honolulu Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: PacificClamChowder Fish Chowder: OpakapakaFishChowder Minnestrone: Minnestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			'Testing Honolulu soup buffet');
	}

}

File: clojure/test/patterns/creational/factory_method_test.clj
(ns patterns.creational.factory_method-test
	(:require 
		[clojure.test :refer :all]
		[clojure.string :as str ]
		[patterns.creational.factory_method :refer :all]
		[common.soups :refer :all]
		[common.soupbuffet :refer :all]
	)
	(:import
		(patterns.creational.factory_method bostonsoupfactorymethod honolulusoupfactorymethod)
		(common.soupbuffet soupbuffet))
)

(defn init-soupbuffet [fm]
	(soupbuffet. 
		(make-buffetname fm)
		 (make-chickensoup fm)
		 (make-clamchowder  fm)
		 (make-fishchowder  fm)
		 (make-minnestrone  fm)
		 (make-pastafazul  fm)
		 (make-tofusoup  fm)
		 (make-vegetablesoup  fm)))
		

(deftest creational-factory-method-test
   (let [
		 boston-soupbuffet (init-soupbuffet (bostonsoupfactorymethod.))
		 honolu-soupbuffet (init-soupbuffet (honolulusoupfactorymethod.))]

		(testing "Boston soup factory "
			(is (=
				"At the  Boston Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: QuahogChowder Fish Chowder: ScrodFishChowder Minnestrone: Minnestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup"
				(str "At the  " (:name boston-soupbuffet) (todays-soups boston-soupbuffet)))))

		(testing "Honolulu soup factory "
			(is  (=
				"At the  Honolulu Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: PacificClamChowder Fish Chowder: OpakapakaFishChowder Minnestrone: Minnestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup"
				(str "At the  " (:name honolu-soupbuffet) (todays-soups honolu-soupbuffet)))))
	))


